src/IECoreScene/LinkedScene.cpp

```diff
@@ -1324,3 +1408,31 @@ void LinkedScene::hash( HashType hashType, double time, MurmurHash &h ) const
 	}
 }
 
+/// serialise this into the linked scene cache so it can just be loaded directly without having to traverse the entire scene
+IECore::PathMatcherDataPtr LinkedScene::linkLocations() const
```

Comment:
No need to return PathMatcherData here - we can rely on PathMatcher's lazy-copy-on-write and just return a PathMatcher directly. I think we planned to do this for `readSet()` and `writeSet()` too?

Reply:
> I think we planned to do this for readSet() and writeSet() too?

Oh yeah - addressed this in my latest changes.